### PR TITLE
Assign Role API

### PR DESCRIPTION
**Pull request by notSanil**
Description: Added API endpoint and changes for #12. Additionally, changed the designation column from string to Enum type in code.

### CODE REVIEW

File: src/main/java/com/pecacm/backend/repository/UserRepository.java [hunk 1]
```diff
@@ -11,4 +13,13 @@ public interface UserRepository extends JpaRepository<User, Integer> {
...
     Optional<User> findByEmail(String email);
+
+    @Query("SELECT u.designation FROM User u " +
+           "WHERE u.email = :email")
+    Optional<Role> findRoleByEmail(String email);
+
+    @Query("UPDATE User " +
+           "SET designation = :newRole " +
+           "WHERE email = :email")
+    void updateRoleByEmail(String email, Role newRole);
```
Comment (harshjohar) [inline src/main/java/com/pecacm/backend/repository/UserRepository.java:24]: Shouldn't there be `@Modifying` annotation present here to modify state of db.

**Review by harshjohar — CHANGES_REQUESTED** [review]
I am unsure about the `@Modifying` part, rest it looks good to me :)

File: src/main/java/com/pecacm/backend/services/UserService.java [hunk 2]
```diff
@@ -57,4 +61,28 @@ public User verifyUser(UUID tokenId) {
...
+        Boolean isUserAuthorizedToChangeRole = requesterRole.equals(Role.Core) || requesterRole.equals(Role.Admin);
+        Boolean isRequestUserRoleLessThanRequester = requestUserRole.compareTo(requesterRole) < 0;
+
+        if (isNewRoleLessThanUserRole && isUserAuthorizedToChangeRole && isRequestUserRoleLessThanRequester)
+        {
+            userRepository.updateRoleByEmail(assignRoleRequest.getRequestEmail(), assignRoleRequest.getNewRole());
+            return Constants.UPDATE_SUCCESS;
+        }
+
+        throw new AcmException(ErrorConstants.USER_UNAUTHORIZED);
```
Comment (ken1000minus7) [inline src/main/java/com/pecacm/backend/services/UserService.java:86]: Add response code 401

**Review by ken1000minus7 — CHANGES_REQUESTED** [review]

**Review by ken1000minus7 — APPROVED** [review]
Lgtm

**Review by harshjohar — APPROVED** [review]
lgtm

**Pull request merged**